src/main/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStub.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.pbis.servicebus;
+
+import com.microsoft.azure.servicebus.IMessage;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.UUID;
+
+
+public class ServiceBusClientStub implements IServiceBusClient {
+
+    private static final ServiceBusClientStub instance = new ServiceBusClientStub();
+    private Queue<IMessage> messagesToReceive = new LinkedList<>();
+
+    private ServiceBusClientStub() {
+        // hiding default constructor
+    }
+
+    public static ServiceBusClientStub getInstance() {
+        return instance;
+    }
+
+    @Override
+    public IMessage receiveMessage() {
+        return messagesToReceive.poll();
+    }
+
+    @Override
+    public void completeMessage(String messageId, UUID messageLockToken) {
+        // nothing to be done
```

Comment:
how about we remove here and `peek` in method above?

Reply:
Normally `receive()` gets the next item from the queue each time and `complete()` prevents it from coming back to the queue when the lock expires. I think the stub is closer to the real thing when it removes messages from the queue in `receive()`.